cmd/lnmuxd/run.go

```diff
@@ -48,6 +48,27 @@ func runAction(c *cli.Context) error {
 		return err
 	}
 
+	group, ctx := errgroup.WithContext(context.Background())
+
+	// Initialise instrumentation server
+	instServer := initInstrumentationServer(cfg.InstrumentationAddress)
+	group.Go(func() error {
+		log.Infow("Instrumentation HTTP server starting",
+			"instrumentationAddress", instServer.Addr)
+
+		return instServer.ListenAndServe()
+	})
+
+	group.Go(func() error {
+		<-ctx.Done()
+
+		// Stop instrumentation server
+		log.Infow("Instrumentation server stopping")
+
+		return instServer.Close()
+	})
+
+	// Setup kubernetes lock
 	releaseLock, err := initDistributedLock(&cfg.DistributedLock)
 	if err != nil {
 		return err
```

Comment:
Another option to enforce clean shutdown in previously-started goroutines is, before this return, to do the following:

```
group.Go(func() error {return err})
```

Reply:
I see what you mean, but the real problem is that the errgroup.Wait() is called only at the end, right?